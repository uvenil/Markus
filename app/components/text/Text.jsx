'use strict';

import React from 'react';

export default class Text extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const theme     = this.props.theme === 'dark' ? require('../../theme.dark.json') : require('../../theme.light.json');
        const textColor = this.props.textColor ? this.props.textColor : theme.primaryTextColor;

        return (
            <span
                className={this.props.className}
                style={{ display : 'block', WebkitUserSelect : 'none', cursor : 'default', fontFamily : this.props.fontFamily, fontWeight : this.props.fontWeight, fontSize : this.props.textSize, color : textColor, pointerEvents : this.props.disabled ? 'none' : 'auto' }}>
                {this.props.children}
            </span>
        );
    }
}

Text.propTypes = {
    className  : React.PropTypes.string,
    fontFamily : React.PropTypes.string,
    fontWeight : React.PropTypes.oneOfType([ React.PropTypes.string, React.PropTypes.number ]),
    textSize   : React.PropTypes.string,
    textColor  : React.PropTypes.string,
    theme      : React.PropTypes.oneOf([ 'light', 'dark' ]),
    disabled   : React.PropTypes.bool
};

Text.defaultProps = {
    className : 'Text',
    theme     : 'light',
    disabled  : false
};

module.exports = Text;
