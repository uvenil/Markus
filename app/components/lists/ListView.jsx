'use strict';

import React from 'react';
import Text from '../text/Text.jsx';
import Unique from '../../utils/Unique';
import Config from '../../../config.json';

export default class ListView extends React.Component {
    constructor(props) {
        super(props);

        this._handleClick = index => {
            if (this.props.onItemClick) {
                this.props.onItemClick(index);
            }
        };

        this._handleRightClick = index => {
            if (this.props.onItemRightClick) {
                this.props.onItemRightClick(index);
            }
        };
    }

    render() {
        const theme = this.props.theme === 'dark' ? require('../../theme.dark.json') : require('../../theme.light.json');

        let backgroundColor         = this.props.backgroundColor ? this.props.backgroundColor : theme.primaryBackgroundColor;
        let selectedBackgroundColor = this.props.selectedBackgroundColor ? this.props.selectedBackgroundColor : theme.selectedBackgroundColor;

        let header;

        if (this.props.header) {
            header = (
                <div style={{ paddingLeft : Config.paddingX1 + 'px', paddingRight : Config.paddingX1 + 'px', paddingTop : Config.paddingX0 + 'px', paddingBottom : Config.paddingX0 + 'px' }}>
                    <Text
                        className="ListViewHeader"
                        theme={this.props.theme}>{this.props.header}</Text>
                </div>
            );
        }

        return (
            <div
                className={this.props.className}
                style={{ backgroundColor : backgroundColor }}>
                {header}
                {this.props.children.map((child, index) => {
                    return (
                        <div
                            key={Unique.elementId()}
                            style={{ backgroundColor : index === this.props.selectedIndex ? selectedBackgroundColor : backgroundColor }}
                            onMouseDown={event => {
                                if (event.nativeEvent.button === 2) {
                                    this._handleRightClick(index);
                                } else {
                                    this._handleClick(index);
                                }
                            }}>
                            {child}
                        </div>
                    );
                })}
            </div>
        );
    }
}

ListView.propTypes = {
    header                  : React.PropTypes.node,
    selectedIndex           : React.PropTypes.number,
    className               : React.PropTypes.string,
    fontFamily              : React.PropTypes.string,
    textSize                : React.PropTypes.string,
    textColor               : React.PropTypes.string,
    selectedTextColor       : React.PropTypes.string,
    backgroundColor         : React.PropTypes.string,
    selectedBackgroundColor : React.PropTypes.string,
    theme                   : React.PropTypes.oneOf([ 'light', 'dark' ]),
    disabled                : React.PropTypes.bool,
    onItemClick             : React.PropTypes.func,
    onItemRightClick        : React.PropTypes.func
};

ListView.defaultProps = {
    selectedIndex : -1,
    className     : 'ListView',
    theme         : 'light',
    disabled      : false
};

module.exports = ListView;
