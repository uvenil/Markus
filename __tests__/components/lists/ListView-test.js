'use strict';

import React from 'react';
import Renderer from 'react-test-renderer';
import ListView from '../../../app/components/lists/ListView.jsx';
import Text from '../../../app/components/text/Text.jsx';

jest.mock('../../../app/utils/Unique');

it('renders correctly', () => {
    const listView = Renderer.create(
        <ListView
            header="Section"
            backgroundColor="#f1f2f4">
            <Text>Item 1</Text>
            <Text>Item 2</Text>
            <Text>Item 3</Text>
        </ListView>
    ).toJSON();

    expect(listView).toMatchSnapshot();
});
