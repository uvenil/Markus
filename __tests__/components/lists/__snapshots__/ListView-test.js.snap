exports[`test renders correctly 1`] = `
<div
  className="ListView"
  style={
    Object {
      "backgroundColor": "#f1f2f4",
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": "4px",
        "paddingLeft": "8px",
        "paddingRight": "8px",
        "paddingTop": "4px",
      }
    }>
    <span
      className="ListViewHeader"
      style={
        Object {
          "WebkitUserSelect": "none",
          "color": "#000",
          "cursor": "default",
          "display": "block",
          "fontFamily": undefined,
          "fontSize": undefined,
          "fontWeight": undefined,
          "pointerEvents": "auto",
        }
      }>
      Section
    </span>
  </div>
  <div
    onMouseDown={[Function]}
    style={
      Object {
        "backgroundColor": "#f1f2f4",
      }
    }>
    <span
      className="Text"
      style={
        Object {
          "WebkitUserSelect": "none",
          "color": "#000",
          "cursor": "default",
          "display": "block",
          "fontFamily": undefined,
          "fontSize": undefined,
          "fontWeight": undefined,
          "pointerEvents": "auto",
        }
      }>
      Item 1
    </span>
  </div>
  <div
    onMouseDown={[Function]}
    style={
      Object {
        "backgroundColor": "#f1f2f4",
      }
    }>
    <span
      className="Text"
      style={
        Object {
          "WebkitUserSelect": "none",
          "color": "#000",
          "cursor": "default",
          "display": "block",
          "fontFamily": undefined,
          "fontSize": undefined,
          "fontWeight": undefined,
          "pointerEvents": "auto",
        }
      }>
      Item 2
    </span>
  </div>
  <div
    onMouseDown={[Function]}
    style={
      Object {
        "backgroundColor": "#f1f2f4",
      }
    }>
    <span
      className="Text"
      style={
        Object {
          "WebkitUserSelect": "none",
          "color": "#000",
          "cursor": "default",
          "display": "block",
          "fontFamily": undefined,
          "fontSize": undefined,
          "fontWeight": undefined,
          "pointerEvents": "auto",
        }
      }>
      Item 3
    </span>
  </div>
</div>
`;
